substrate/frame/contracts/proc-macro/src/lib.rs

```diff
@@ -206,9 +207,25 @@ impl HostFn {
 					}
 					not_deprecated = false;
 				},
+				"mutable" => {
+					if mutable {
+						return Err(err(span, "#[mutable] can only be specified once"))
+					}
+					mutable = true;
+				},
 				_ => return Err(err(span, msg)),
 			}
 		}
+
+		if mutable {
```

Comment:
Seems inverted. Shouldn't it be caught by your tests.

```suggestion
		if !mutable {
```